WordPress/gutenberg
# Icons: Update Pin icon, add new Post icon.

## Description

This PR updates the Pin icon (the existing one is a dashicon) to look like this:

<img width="393" alt="Screenshot 2022-03-01 at 14 45 49" src="https://user-images.githubusercontent.com/1204802/156181258-48ad35bc-6a10-45c5-9420-f73eb53f75f5.png">

It also adds a new icon, "Post", which is to indicate the post type of _post_ (should it be called something else?):

<img width="375" alt="Screenshot 2022-03-01 at 14 46 07" src="https://user-images.githubusercontent.com/1204802/156181363-12b479b5-3b0c-40bd-b235-edda971c43c2.png">

## Testing Instructions

Please check these in storybook, check that they look the same as the screenshots above.

The icons will be used in [this PR](https://github.com/WordPress/gutenberg/pull/29172#issuecomment-1046672202), and the Figma Design Library has been updated.

## Checklist:
